File: Data Structures & Algorithms/single-element-in-a-sorted-array/submission-4.py
class Solution:
    def singleNonDuplicate(self, nums: List[int]) -> int:
        l, r = 0, len(nums) - 1
        m = (l + r) // 2
        while l < r:
            print("here")
            print(l)
            print(r)
            print(m)
            if m%2==0 and nums[m] == nums[m-1]:
                r = m - 1
            elif m%2==0 and nums[m] == nums[m+1]:
                l = m + 1
            elif m%2==0:
                return nums[m]
            elif m%2 and nums[m] == nums[m-1]:
                l = m + 1
            elif m%2 and nums[m] == nums[m+1]:
                r = m - 1
            else:
                return nums[m]
            m = (l + r) // 2
        return nums[m]

File: Data Structures & Algorithms/concatenation-of-array/submission-4.ts
class Solution {
    /**
     * @param {number[]} nums
     * @return {number[]}
     */
    getConcatenation(nums: number[]): number[] {
        let ans = [];
        let n = nums.length
        for (let i = 0; i < n; i++) {
            ans[i] = ans[i+n] = nums[i]
        }
        return ans
    }
}
